src/GitHub.UI/Controls/DropDownActionLinkBehaviour.cs

```diff
@@ -0,0 +1,58 @@
+﻿using System;
+using System.Threading;
+using System.Windows;
+using System.Windows.Controls;
+using System.Windows.Controls.Primitives;
+using System.Windows.Interactivity;
+
+namespace GitHub.UI
+{
+    public class DropDownActionLinkBehaviour : Behavior<Button>
+    {
+        private long attachedCount;
+        private bool isContextMenuOpen;
+
+        protected override void OnAttached()
+        {
+            base.OnAttached();
+            AssociatedObject.AddHandler(Button.ClickEvent, new RoutedEventHandler(AssociatedObject_Click), true);
+        }
+
+        void AssociatedObject_Click(object sender, System.Windows.RoutedEventArgs e)
+        {
+            Button source = sender as Button;
+            if (source != null && source.ContextMenu != null)
+            {
+                // Only open the ContextMenu when it is not already open. If it is already open,
+                // when the button is pressed the ContextMenu will lose focus and automatically close.
+                if (!isContextMenuOpen)
+                {
+                    source.ContextMenu.AddHandler(ContextMenu.ClosedEvent, new RoutedEventHandler(ContextMenu_Closed), true);
+                    Interlocked.Increment(ref attachedCount);
```

Comment:
Good question, this is copied over straight from the Desktop app, haha!
I'm actually going to :fire: this since I have a better way of doing it without having to use Context Menus, so there 😄